packages/react/src/components/DataTable/__tests__/TableSelectRow-test.js

```diff
@@ -23,45 +23,49 @@ describe('DataTable.TableSelectRow', () => {
     };
   });
 
-  it('should render', () => {
-    const wrapper = mount(
-      <Table>
-        <TableHead>
-          <TableRow>
-            <TableSelectRow {...mockProps} />
-          </TableRow>
-        </TableHead>
-      </Table>
-    );
-    expect(wrapper).toMatchSnapshot();
-  });
+  describe('renders as expected - Component API', () => {
+    it('should render', () => {
+      const { container } = render(
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableSelectRow {...mockProps} />
+            </TableRow>
+          </TableHead>
+        </Table>
+      );
+      expect(container).toMatchSnapshot();
+    });
 
-  it('should render with the provided class name', () => {
-    const customClassName = 'custom-table-select-row-classname';
-    const wrapper = mount(
-      <Table>
-        <TableHead>
-          <TableRow>
-            <TableSelectRow {...mockProps} className={customClassName} />
-          </TableRow>
-        </TableHead>
-      </Table>
-    );
-    const elements = wrapper.find(`td.${customClassName}`);
-    expect(elements.length).toBe(1);
+    it('should render with the provided class name', () => {
+      const customClassName = 'custom-table-select-row-classname';
```

Comment:
It looks like there's already a custom classname provided through `mockProps`.

Honestly I'm an advocate for removing abstractions like `mockProps` in favor of explicitly placing the props on the component in render. To me it feels like a hasty abstraction that reduces readability and doesn't optimize for change.

Reply:
Now using the custom className frpm `mockProps`. I didn't remove 'cause most of the props are required so I would've just duplicated the values in every test. But I'm happy to do that if you feel that makes sense!